### PR TITLE
feat: Update OTEL dependencies to v0.112.0

### PR DESCRIPTION
**Pull request by StefanKurek**
This changes the OTEL dependencies to v0.112.0

### CODE REVIEW

File: processor/agentmetricsprocessor/factory_test.go [hunk 1]
```diff
@@ -21,15 +21,16 @@ import (
...
+	"go.opentelemetry.io/collector/processor/processortest"
 )
 
 func TestCreateDefaultConfig(t *testing.T) {
 	assert.NoError(t, componenttest.CheckConfigStruct(NewFactory().CreateDefaultConfig()))
 }
 
 func TestCreateProcessor(t *testing.T) {
-	mp, err := createMetricsProcessor(context.Background(), processor.CreateSettings{}, createDefaultConfig(), consumertest.NewNop())
+	cfg := createDefaultConfig().(*Config)
```
Comment (braydonk) [inline processor/agentmetricsprocessor/factory_test.go:32]: Could you explain why this was done? I'm having trouble understanding the reason.

**Review by braydonk — COMMENTED** [review]
Most of the changes look fine to me, just want to understand one change.

File: processor/agentmetricsprocessor/factory_test.go (hunk 1 above)
Comment (StefanKurek) [reply processor/agentmetricsprocessor/factory_test.go:32]: @braydonk Honestly I can't remember why I did this for the config...but good catch! Doesn't seem to be necessary. Might have came from a copy/paste while looking at existing factory tests within the latest release.

Pushing a change to just pass in `createDefaultConfig()` in the 3 files where I did this.

File: service/main.go [hunk 2]
```diff
@@ -43,6 +50,17 @@ func MainContext(ctx context.Context) {
 	params := otelcol.CollectorSettings{
 		Factories: components,
 		BuildInfo: info,
+		ConfigProviderSettings: otelcol.ConfigProviderSettings{
+			ResolverSettings: confmap.ResolverSettings{
+				ProviderFactories: []confmap.ProviderFactory{
+					fileprovider.NewFactory(),
```
Comment (ridwanmsharif) [inline service/main.go:56]: Is there a reason we're adding all these providers for our distro? Not sure why these are needed now

Comment (ridwanmsharif) [reply service/main.go:56]: Don't we only use the file provider?

Comment (jefferbrecht) [reply service/main.go:56]: There was an upstream change that forces you do explicitly list providers now, I don't have a reference at hand though

Comment (ridwanmsharif) [reply service/main.go:56]: Ack, it stems from https://github.com/open-telemetry/opentelemetry-collector/pull/9513/files

We used to default to all the default providers so this change makes sense. We should separately reduce this list to just the file provider but that shouldn't block this PR

**Review by ridwanmsharif — APPROVED** [review]

**Pull request merged**